graphqlapi/src/main/java/io/stargate/graphql/schema/graphqlfirst/processor/EntityModel.java

```diff
@@ -155,42 +155,33 @@ public boolean isFederated() {
 
   /**
    * Validates a set of conditions to ensure that they will produce a valid CQL query that doesn't
-   * require ALLOW FILTERING.
+   * require ALLOW FILTERING for this entity.
    *
    * @return an optional error message if the validation failed, empty otherwise.
    */
-  public Optional<String> validateConditions(Collection<WhereConditionModel> conditions) {
+  public Optional<String> validateNoFiltering(Collection<WhereConditionModel> conditions) {
 
     long indexCount = conditions.stream().filter(c -> !c.getField().isPrimaryKey()).count();
 
     if (indexCount == 0) {
-      return validateConditionsNoIndex(conditions);
+      return validateNoFilteringWhenNoIndex(conditions);
     }
     if (indexCount == 1) {
-      return validateConditionsOneIndex(conditions);
+      return validateNoFilteringWhenOneIndex(conditions);
     }
     return Optional.of("cannot use more than one indexed field per operation");
   }
 
-  private Optional<String> validateConditionsNoIndex(Collection<WhereConditionModel> conditions) {
-    // All partition key fields must be restricted by equality relations
+  private Optional<String> validateNoFilteringWhenNoIndex(
+      Collection<WhereConditionModel> conditions) {
+    // All partition key fields must be present
     for (FieldModel field : this.getPartitionKey()) {
-      Optional<WhereConditionModel> maybeCondition =
-          conditions.stream().filter(c -> c.getField().equals(field)).findFirst();
-      if (!maybeCondition.isPresent()) {
+      if (conditions.stream().noneMatch(c -> c.getField().equals(field))) {
         return Optional.of(
             String.format(
                 "every partition key field of type %s must be present (expected: %s).",
                 this.getGraphqlName(), describePartitionKey()));
       }
-      WhereConditionModel condition = maybeCondition.get();
-      if (condition.getPredicate() != Predicate.EQ && condition.getPredicate() != Predicate.IN) {
-        return Optional.of(
-            String.format(
-                "the only predicates allowed for partition key fields are EQ and IN "
-                    + "(got %s for %s).",
-                condition.getPredicate(), field.getGraphqlName()));
-      }
```

Comment:
Removed this because static checks on predicates are now all done at deployment time in `WhereConditionModelBuilder`.

As a reminder, `validateNoFiltering` is invoked both at deployment time, and every time a query is executed (because if the arguments are optional, omitting some of them changes the set of conditions).